pkg/operator/operator.go

```diff
@@ -265,35 +269,40 @@ func isMachineWebhook(obj interface{}) bool {
 }
 
 func (optr *Operator) worker() {
-	for optr.processNextWorkItem() {
+	var cancel context.CancelFunc
+	processNext := true
+	optr.context, cancel = context.WithCancel(context.Background())
+	for processNext {
+		processNext, cancel = optr.processNextWorkItem(cancel)
 	}
+	cancel()
 }
 
-func (optr *Operator) processNextWorkItem() bool {
+func (optr *Operator) processNextWorkItem(cancel context.CancelFunc) (bool, context.CancelFunc) {
 	key, quit := optr.queue.Get()
 	if quit {
-		return false
+		return false, cancel
 	}
 	defer optr.queue.Done(key)
 
 	glog.V(4).Infof("Processing key %s", key)
-	err := optr.syncHandler(key.(string))
-	optr.handleErr(err, key)
+	// Cancel previous reconcile
+	cancel()
+	optr.context, cancel = context.WithTimeout(context.Background(), maoSyncTimeout)
+	go func() {
```

Comment:
Why is this running in a subroutine??
The code this PR is changing is pretty standard controller skeleton and we have no intent to change it.

If we needed async/multiple calls we could run just more workers https://github.com/openshift/machine-api-operator/blob/c9a1f7584b4bd025e1e99444d0f6281c86a18a1a/cmd/machine-api-operator/start.go#L132 but I see no need for it.

If anything the reconciling loop could be modified to reconcile the different resources here asynchronously https://github.com/openshift/machine-api-operator/blob/0ce2439a9f0ec28dbc45987dc5c28b223e2cfd05/pkg/operator/sync.go#L44
But also I don't see a real need for that. We'll only introduce complexity here is there's a really strong argument in favour of it.